### ISSUE TITLE
Fatal compile error when building with CUDA support

### ISSUE THREAD

**Issue by shawnmasters** [AlexeyAB/darknet#2563]
I get a lot of warnings whenever I build the code for CPU only support, but it does complete and link an executable for Linux.  When I turn on GPU or CUDNN support I get errors in the build.  for CUDNN it is the following.

```In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/stddef.h:217:0:
/usr/local/cuda/include/crt/host_runtime.h:19:2: warning: #warning "crt/device_functions.h is an internal header file and must not be used directly.  Please use cuda_runtime_api.h or cuda_runtime.h instead." [-Wcpp]
 #warning "crt/device_functions.h is an internal header file and must not be used directly.  Please use cuda_runtime_api.h or cuda_runtime.h instead.
  ^
In file included from /tmp/tmpxft_0000531d_00000000-5_im2col_kernels.compute_61.cudafe1.stub.c:5:0,
                 from tmpxft_0000531d_00000000-5_im2col_kernels.compute_61.cudafe1.stub.c:1:
/usr/local/cuda/include/crt/host_runtime.h:19:2: warning: #warning "crt/device_functions.h is an internal header file and must not be used directly.  Please use cuda_runtime_api.h or cuda_runtime.h instead." [-Wcpp]
 #warning "crt/device_functions.h is an internal header file and must not be used directly.  Please use cuda_runtime_api.h or cuda_runtime.h instead.
  ^
/tmp/tmpxft_0000531d_00000000-5_im2col_kernels.compute_61.cudafe1.stub.c: In function ‘void __device_stub__Z17im2col_gpu_kerneliPKfiiiiiiiPf(int, const float*, int, int, int, int, int, int, int, float*)’:
/tmp/tmpxft_0000531d_00000000-5_im2col_kernels.compute_61.cudafe1.stub.c:25:235: error: ‘__args_arr’ was not declared in this scope
 t int __par2, const int __par3, const int __par4, const int __par5, const int __par6, const int __par7, const int __par8, float *__par9){__cudaSetup
                                                                                                                                          ^
compilation terminated due to -Wfatal-errors.
Makefile:145: recipe for target 'obj/im2col_kernels.o' failed
make: *** [obj/im2col_kernels.o] Error 1
```
I have CUDA and CUDNN installed and can compile the original darknet version without any problems.  What am I missing?

**Comment by AlexeyAB** [comment]
@shawnmasters Hi,

You didn't install cuda properly. During CUDA installation you should press yes for `create symlink`.
The installer will prompt to see if you want to create the **symlink** for `/usr/local/cuda` -> `/usr/local/cuda-10.0` https://docs.nvidia.com/cuda/cuda-installation-guide-mac-os-x/index.html#install

* Read this: https://github.com/AlexeyAB/darknet/issues/2290#issuecomment-457525243
* Or this: https://github.com/pjreddie/darknet/issues/200#issuecomment-329692411

----

* Do you use `make` or `cmake` for compiling?
* What params did you use in the Makefile?
* What OS do you use?
* Can you show output of commands?
```
nvcc --version
nvidia-smi
```

* Did you try?
```
make clean
make
```

It looks like this error very similar to: https://github.com/pjreddie/darknet/issues/1246#issuecomment-451275558

**Comment by shawnmasters** [comment]
I used the prepackaged versions of CUDA from NVidia (one of the boxes is a DGX-1 and I have to use NVidia's docker builds).  As far as the synlink, it's there so I don't think that is the problem.

While I was able to compile the darknet code from the the original site, this repo was giving me these errors.  I noticed that `nvcc --version` gives me a version of 9, not 10.  I found there was a system install one that is being called despite the /usr/local/cuda/bin being in the front of the path.

Changing the NVCC variable in the makefile seems to have fixed this issue.

Thanks
